rbf-chunker/src/main.rs

```diff
@@ -0,0 +1,227 @@
+use std::fs::File;
+use std::io::prelude::*;
+use std::io::{Error, ErrorKind};
+use std::path::{Path, PathBuf};
+
+#[derive(Debug, Copy, Clone)]
+enum SectionType {
+    Unkown,
+    Header,
+    Footer,
+    Packet { crc: [u8; 2] },
+    PacketCRC,
+}
+
+#[derive(Debug, Copy, Clone)]
+struct SectionInfo {
+    offset: usize,
+    size: usize,
+    stype: SectionType,
+}
+
+impl SectionInfo {
+    fn new_header(size: usize) -> Self {
+        SectionInfo {
+            offset: 0,
+            size,
+            stype: SectionType::Header,
+        }
+    }
+
+    fn new_unkown(offset: usize, size: usize) -> Self {
+        SectionInfo {
+            offset,
+            size,
+            stype: SectionType::Unkown,
+        }
+    }
+
+    fn new_footer(offset: usize, size: usize) -> Self {
+        SectionInfo {
+            offset,
+            size,
+            stype: SectionType::Footer,
+        }
+    }
+
+    fn new_packet(offset: usize, size: usize, crc: [u8; 2]) -> Self {
+        SectionInfo {
+            offset,
+            size,
+            stype: SectionType::Packet { crc },
+        }
+    }
+
+    fn new_packet_crc(offset: usize) -> Self {
+        SectionInfo {
+            offset,
+            size: 2,
+            stype: SectionType::PacketCRC,
+        }
+    }
+}
+
+const RBF_SIZE: usize = 7020944;
```

Comment:
This value changes depending on the target device; it's not a constant.

Reply:
Ah, okay. All usages replaced with len.

Reply:
Are the footer/header sizes also variable between target devices?

Reply:
The docs mention that the settings are `0x80`-`0x127`, which leads me to think the header is constant at least.